File: Desktop/mediflow_pro/backend-server/backend/app/models/ad.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum

class AdType(str, Enum):
    BANNER = "banner"
    POPUP = "popup"
    VIDEO = "video"

class Ad(BaseModel):
    id: Optional[str] = None
    title: str
    image_url: str
    target_url: Optional[str] = None
    ad_type: AdType = AdType.BANNER
    is_active: bool = True
    priority: int = 0  # Higher number = higher priority
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    clinic_id: Optional[str] = None  # None means all clinics
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        json_schema_extra = {
            "example": {
                "title": "Special Offer",
                "image_url": "https://example.com/ad.jpg",
                "target_url": "https://example.com/offer",
                "ad_type": "banner",
                "is_active": True,
                "priority": 1,
                "clinic_id": None
            }
        }

File: Desktop/mediflow_pro/mobile-app/lib/models/ad.dart
class Ad {
  final String? id;
  final String title;
  final String imageUrl;
  final String? targetUrl;
  final String adType;
  final bool isActive;
  final int priority;
  final String? clinicId;
  final DateTime? startDate;
  final DateTime? endDate;
  final String? localAssetPath; // NEW: Support for local assets

  Ad({
    this.id,
    required this.title,
    this.imageUrl = '',
    this.targetUrl,
    this.adType = 'banner',
    this.isActive = true,
    this.priority = 0,
    this.clinicId,
    this.startDate,
    this.endDate,
    this.localAssetPath, // NEW
  });

  factory Ad.fromJson(Map<String, dynamic> json) {
    return Ad(
      id: json['id'],
      title: json['title'] ?? '',
      imageUrl: json['image_url'] ?? '',
      targetUrl: json['target_url'],
      adType: json['ad_type'] ?? 'banner',
      isActive: json['is_active'] ?? true,
      priority: json['priority'] ?? 0,
      clinicId: json['clinic_id'],
      startDate: json['start_date'] != null 
          ? DateTime.parse(json['start_date']) 
          : null,
      endDate: json['end_date'] != null 
          ? DateTime.parse(json['end_date']) 
          : null,
      localAssetPath: json['local_asset_path'], // NEW
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'image_url': imageUrl,
      'target_url': targetUrl,
      'ad_type': adType,
      'is_active': isActive,
      'priority': priority,
      'clinic_id': clinicId,
      'start_date': startDate?.toIso8601String(),
      'end_date': endDate?.toIso8601String(),
      'local_asset_path': localAssetPath, // NEW
    };
  }
}
